Execinfo can be required on Linux too

**Affects versions :**
 - OS: Alpine Linux Edge. Uses musl libc
 - Version of Xournal++: v1.0.13

**Describe the bug**

Xournalpp just assumes that linux always has execinfo installed, however this is not necessarily the case, only glibc bundles it. You still have to install libexecinfo on musl libc (and possibly other libcs? Not quite sure on that). It'd be nice if xournalpp just checked for the existence of `execinfo.h` instead of guessing. In case execinfo isn't built-into libc on linux it also needs to be linked into the `xournalpp` binary, otherwise it can't find the required symbols:

```
[100%] Linking CXX executable xournalpp
/usr/lib/gcc/x86_64-alpine-linux-musl/8.3.0/../../../../x86_64-alpine-linux-musl/bin/ld: util/libutil.a(CrashHandler.cpp.o): in function `crashHandler(int)':
/home/builder/aports/testing/xournalpp/src/xournalpp-1.0.12/src/util/CrashHandlerUnix.h:70: undefined reference to `backtrace'
/usr/lib/gcc/x86_64-alpine-linux-musl/8.3.0/../../../../x86_64-alpine-linux-musl/bin/ld: /home/builder/aports/testing/xournalpp/src/xournalpp-1.0.12/src/util/CrashHandlerUnix.h:88: undefined reference to `backtrace_symbols'
/usr/lib/gcc/x86_64-alpine-linux-musl/8.3.0/../../../../x86_64-alpine-linux-musl/bin/ld: util/libutil.a(Stacktrace.cpp.o): in function `Stacktrace::printStracktrace(std::ostream&)':
/home/builder/aports/testing/xournalpp/src/xournalpp-1.0.12/src/util/Stacktrace.cpp:82: undefined reference to `backtrace'
/usr/lib/gcc/x86_64-alpine-linux-musl/8.3.0/../../../../x86_64-alpine-linux-musl/bin/ld: /home/builder/aports/testing/xournalpp/src/xournalpp-1.0.12/src/util/Stacktrace.cpp:83: undefined reference to `backtrace_symbols'
collect2: error: ld returned 1 exit status
make[2]: *** [src/CMakeFiles/xournalpp.dir/build.make:593: src/xournalpp] Error 1
make[1]: *** [CMakeFiles/Makefile2:233: src/CMakeFiles/xournalpp.dir/all] Error 2
make: *** [Makefile:130: all] Error 2
```

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -49,15 +49,9 @@ else ()
     set(LIB_ENDING "so")
 endif ()
 
-if (WIN32)
-    set (xournalpp_LDFLAGS ${xournalpp_LDFLAGS} "-mwindows")
-endif ()
-
 ## For touch workaround, may need to be disabled for a Wayland Build
-if (${CMAKE_SYSTEM_NAME} MATCHES "Linux")
+if (${CMAKE_SYSTEM_NAME} MATCHES "Linux" OR ${CMAKE_SYSTEM_NAME} MATCHES "FreeBSD")
     set (xournalpp_LDFLAGS ${xournalpp_LDFLAGS} "-lX11 -lXi")
-elseif(${CMAKE_SYSTEM_NAME} MATCHES "FreeBSD")
-	set (xournalpp_LDFLAGS ${xournalpp_LDFLAGS} "-lX11 -lXi -lexecinfo")
 endif ()
 
 ## Libraries ##
@@ -66,6 +60,14 @@ macro (add_includes_ldflags LDFLAGS INCLUDES)
     set (xournalpp_LDFLAGS ${xournalpp_LDFLAGS} "${LDFLAGS}")
     set (xournalpp_INCLUDE_DIRS ${xournalpp_INCLUDE_DIRS} "${INCLUDES}")
 endmacro (add_includes_ldflags LDFLAGS INCLUDES)
+
+# libexec
+if (WIN32)
+    set (xournalpp_LDFLAGS ${xournalpp_LDFLAGS} "-mwindows")
+else ()
+    find_package(Backtrace REQUIRED)
+    add_includes_ldflags ("${Backtrace_LIBRARIES}" "${Backtrace_INCLUDE_DIRS}")
+endif ()
 
 # GLIB
 pkg_check_modules (Glib REQUIRED "glib-2.0 >= 2.32.0")

